lib/Db/TimelineQueryFaceRecognition.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\Memories\Db;
+
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\Files\Folder;
+use OCP\IDBConnection;
+
+trait TimelineQueryFaceRecognition
+{
+    protected IDBConnection $connection;
+
+    public function transformFaceRecognitionFilter(IQueryBuilder &$query, string $userId, string $personStr)
+    {
+        // Get title and uid of face user
+        $personNames = explode('/', $personStr);
+        if (2 !== \count($personNames)) {
+            throw new \Exception('Invalid person query');
+        }
+
+        $personUid = $personNames[0];
+        $personName = $personNames[1];
+
+        // Join with images
+        $query->innerJoin('m', 'facerecog_images', 'fri',
+            $query->expr()->eq('fri.file', 'm.fileid')
+        );
+
+        // Join with faces
+        $query->innerJoin('fri', 'facerecog_faces', 'frf',
+            $query->expr()->eq('frf.image', 'fri.id')
```

Comment:
Similar question - assuming these have the same type?

Reply:
Here both types are equal... 
Though ideally maybe I should change them to bigint as well... Maybe later. It is not necessary here.